controllers/ingressnodefirewallconfig_controller.go

```diff
@@ -161,6 +178,66 @@ func (r *IngressNodeFirewallConfigReconciler) syncIngressNodeFwConfigResources(c
 			if len(config.Spec.NodeSelector) > 0 {
 				ds.Spec.Template.Spec.NodeSelector = config.Spec.NodeSelector
 			}
+			daemonContainer := 0
+			for idx, c := range ds.Spec.Template.Spec.Containers {
+				if c.Name == daemonContainerName {
+					daemonContainer = idx
+					break
+				}
+			}
+
+			ds.Spec.Template.Spec.Containers[daemonContainer].SecurityContext = &corev1.SecurityContext{
+				Privileged: ptr.To[bool](true),
+				RunAsUser:  ptr.To[int64](0),
+				Capabilities: &corev1.Capabilities{
+					Add: []corev1.Capability{
+						"CAP_BPF",
+						"CAP_NET_ADMIN",
+					},
+				},
+			}
+
+			if useBPFMan {
+				ds.Spec.Template.Spec.Containers[daemonContainer].VolumeMounts = append(
```

Comment:
There is a risk of accessing `ds.Spec.Template.Spec.Containers[daemonContainer]` out of bounds if the `daemonContainerName` is not found. It would be safer to check if `daemonContainer` has been updated from its initial value before using it to access an array element... something like:

```go
daemonContainer := -1 // Start with an invalid index
for idx, c := range ds.Spec.Template.Spec.Containers {
    if c.Name == daemonContainerName {
        daemonContainer = idx
        break
    }
}

if daemonContainer == -1 {
    logger.Error(nil, "No container found with the specified name", "ContainerName", daemonContainerName)
    return fmt.Errorf("no container found with the specified name: %s", daemonContainerName)
}
```